--- tce-load.orig	2015-11-17 09:33:16.000000000 +0900
+++ tce-load	2015-11-17 09:40:06.000000000 +0900
@@ -94,14 +94,15 @@
 
 copyInstall() {
 	[ -d /mnt/test ] || sudo mkdir -p /mnt/test
-	sudo mount $1 /mnt/test -t squashfs -o loop,ro,bs=4096
+	sudo /usr/local/bin/unsquashfs -f -d /mnt/test $1
 	if [ "$?" == 0 ]; then
 		if [ "$(ls -A /mnt/test)" ]; then
 			yes "$FORCE" | sudo cp -ai /mnt/test/. / 2>/dev/null
 			[ `find /mnt/test/ -type d -name modules` ] && MODULES=TRUE
 		fi
-		sudo umount -d /mnt/test
+		sudo rm -rf /mnt/test
 	fi
+	[ "$BOOTING" ] || [ -d /mnt/test ] && rmdir /mnt/test
 }
 
 update_system() {
